### PR TITLE
Add proof harness for aws_hash_byte_cursor_ptr function

### PR DESCRIPTION
**Pull request by feliperodri**
Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>

*Description of changes:*
Adds a proof harness for `aws_hash_byte_cursor_ptr` function. However, the proof depends on https://github.com/awslabs/aws-c-common/pull/364.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by feliperodri** [comment]
> It looks like the check failed?

Yes, we are waiting on a fix from CBMC to solve this issue.

File: .cbmc-batch/jobs/aws_hash_byte_cursor_ptr/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# limitations under the License.
+
+###########
+MAX_CURSOR_SIZE ?= 16
+DEFINES += -DMAX_CURSOR_SIZE=$(MAX_CURSOR_SIZE) 
+
+UNWINDSET += hashlittle2.2:$(shell echo $$(( $$(( $(MAX_CURSOR_SIZE) / 12 )) +1 )) )
+UNWINDSET += __CPROVER_file_local_lookup3_c_hashlittle2.0:$(shell echo $$(( $(MAX_CURSOR_SIZE) - 11)))
+UNWINDSET += __CPROVER_file_local_lookup3_c_hashlittle2.1:$(shell echo $$(( $(MAX_CURSOR_SIZE) - 11)))
+UNWINDSET += __CPROVER_file_local_lookup3_c_hashlittle2.2:$(shell echo $$(( $(MAX_CURSOR_SIZE) - 11)))
```
Comment (angelhof) [inline .cbmc-batch/jobs/aws_hash_byte_cursor_ptr/Makefile]: Similarly to the comments at #364

**Comment by feliperodri** [comment]
@karkhaz could you review the proof harnesses?

File: include/aws/common/private/lookup3.c [hunk 2]
```diff
@@ -322,12 +327,16 @@ static uint32_t hashlittle( const void *key, size_t length, uint32_t initval)
...
-     * noticably faster for short strings (like English words).
+     * rest of the string. Every machine with memory protection I've seen
+     * does it on word boundaries, so is OK with this. But VALGRIND and CBMC
+     * will still catch it and complain. CBMC will ignore this type of error
+     * in the code block between the pragmas "CPROVER check push" and
+     * "CPROVER check pop". The masking trick does make the hash noticably
+     * faster for short strings (like English words).
      */
 #ifndef VALGRIND
+#pragma CPROVER check push
```
Comment (karkhaz) [inline include/aws/common/private/lookup3.c]: How come you guard these pragmas with `#ifdef CBMC` above, but not here?

File: include/aws/common/private/lookup3.c [hunk 3]
```diff
@@ -506,12 +516,16 @@ static void hashlittle2(
...
+#pragma CPROVER check disable "pointer"
```
Comment (ColdenCullen) [inline include/aws/common/private/lookup3.c]: guard with `#ifdef CBMC` please

File: include/aws/common/private/lookup3.c (hunk 2 above)
Comment (feliperodri) [reply include/aws/common/private/lookup3.c]: Done! You must merge first https://github.com/awslabs/aws-c-common/pull/364

File: include/aws/common/private/lookup3.c (hunk 3 above)
Comment (feliperodri) [reply include/aws/common/private/lookup3.c]: Done! You must merge first https://github.com/awslabs/aws-c-common/pull/364

**Comment by feliperodri** [comment]
@ColdenCullen I've implement your comments, could you review `lookup3.c` and `hash_table.c`?

**Comment by feliperodri** [comment]
@ColdenCullen @karkhaz could you merge it?

**Pull request merged**